gyp/pylib/gyp/input.py

```diff
@@ -1222,6 +1222,8 @@ def ProcessVariablesAndConditionsInDict(the_dict, phase, variables_in,
     # Skip "variables", which was already processed if present.
     if key != 'variables' and type(value) is str:
       expanded = ExpandVariables(value, phase, variables, build_file)
+      if type(expanded) in (long, list, dict):
+        continue
```

Comment:
Seems I answered that question 8 months ago in https://github.com/nodejs/node-gyp/issues/903#issuecomment-209012970...  I suppose the question then becomes whether this is the only ExpandVariables call site that needs updating.